### PR TITLE
One more time fix the AssetType

### PR DESCRIPTION
**Pull request by YohannParis**
# Description

* Allow asset names to be updated on the ProjectAssets database.
* Change AssetTypes from uppercase to kebab-case

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java [hunk 1]
```diff
@@ -1,24 +1,57 @@
...
+	@JsonProperty("publication")
+	PUBLICATION,
+
+	@JsonProperty("notebook-session")
+	NOTEBOOK_SESSION,
 	;
 
-	@JsonValue
-	private final String value;
+	public static AssetType getAssetType(final String assetTypeName, final ObjectMapper objectMapper) throws ResponseStatusException {
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java:47]: Why do we need to pass `objectMapper`, can we not instantiate one in this method?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java [hunk 2]
```diff
@@ -1,24 +1,57 @@
...
+		try {
+			return objectMapper.convertValue(assetTypeName, AssetType.class);
+		} catch (final IllegalArgumentException iae) {
+			log.error("Error creating project assets, invalid AssetType", iae);
+			throw new ResponseStatusException(
+					HttpStatus.BAD_REQUEST,
+					"Failed to create project asset");
```
Comment (YohannParis) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java:54]:
```suggestion
			log.error("Error converting the string assetTypeName, into a valid AssetType", iae);
			throw new ResponseStatusException(
					HttpStatus.BAD_REQUEST,
					"Failed to convert a AssetTypeName into a AssetType");
```

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java (hunk 1 above)
Comment (bigglesandginger) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java:47]: Enums cannot Autowire/inject beans.  The main workarounds for this is to have a factory class that provides the enum and can do the heavy lifting.  Will that be better?

Comment (YohannParis) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/AssetType.java:47]: No, all good, thanks

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServices.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	private final WorkflowService workflowService;
+
+	/**
+	 * Get the service for a given asset type
+	 * @param type The type of asset to get the service for
+	 * @return The service for the given asset type
+	 */
+	public TerariumAssetService<? extends TerariumAsset> getServiceByType(final AssetType type) {
+        return switch (type) {
+            case ARTIFACT -> artifactService;
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServices.java:28]: I have nothing but good things to say about this new syntax for `switch` statements :)

File: packages/client/hmi-client/src/page/project/components/tera-project-overview.vue [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <template>
 	<tera-asset
 		:name="useProjects().activeProject.value?.name"
-		:authors="useProjects().activeProject.value?.authors?.join(', ')"
+		:authors="useProjects().activeProject.value?.authors?.sort().join(', ')"
```
Comment (bigglesandginger) [inline packages/client/hmi-client/src/page/project/components/tera-project-overview.vue:4]: I hope that sorts by last name... 🤔

**Review by bigglesandginger — APPROVED** [review]

**Review by dvince2 — APPROVED** [review]

File: packages/client/hmi-client/src/page/project/components/tera-project-overview.vue (hunk 4 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/page/project/components/tera-project-overview.vue:4]: Nope, it does not, I was messing around and forgot to change that, not really important to be honest.

**Comment by YohannParis** [comment]
@dvince2 I have an issue where Model do not get their AssetName save when I create a ProjectAssetType... which makes no sense to me.

**Comment by YohannParis** [comment]
I think I want to wait until after @pascaleproulx demo on Tuesday to merge this PR. Works great locally, but this will be harder to reverse properly or quickly enough.

**Review by dgauldie — APPROVED** [review]
LGTM. Tested as much as I could.

Fingers crossed

**Pull request merged**